### PR TITLE
[vector.bool] Excise use of undefined term 'conversion operator'

### LINKED ISSUE #444: "conversion operator" denotes functions or operators?
[vector.bool] §23.3.7/4 refers to `std::vector<bool>::reference::operator bool` as a conversion operator, whereas it's a conversion function. The index indicates that "conversion operator" is the same as "conversion, user-defined." However [expr.cast] §5.4/2 uses the term to refer to static_cast, const_cast, dynamic_cast and reinterpret_cast, which seems more appropriate.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #444.

### CODE REVIEW

File: source/compatibility.tex [hunk 1]
```diff
@@ -1289,7 +1289,7 @@
 \ref{istream::sentry},
 \ref{ostream::sentry},
 \ref{iostate.flags}
-\change Specify use of explicit in existing boolean conversion operators
+\change Specify use of explicit in existing boolean conversion functions
```
Comment (tkoeppe) [inline source/compatibility.tex]: Should "explicit" appear in code font, since it's actually a keyword?

Comment (jensmaurer) [inline source/compatibility.tex]: Yes, but I shouldn't make that change here, which deals with "conversion operator" vs. "conversion function".

I've created #1019 to deal with the "explicit" code font issue.

Comment (tkoeppe) [inline source/compatibility.tex]: OK, sure, but I would have had no problem with a minor drive-by fix like that if this is a single, isolated problem. You can stick it into a separate commit in the same pull request, and I could retain the separate commits (i.e. not squash them), so we still get nicely atomic commits.
